File: web/src/api/version.test.ts
import { describe, it, expect, vi } from 'vitest'
import { fetchVersion } from './version'

describe('fetchVersion', () => {
  it('returns parsed PublicVersion', async () => {
    globalThis.fetch = vi.fn().mockResolvedValue({
      ok: true,
      json: async () => ({ apiVersion: '1.1.0', skillVersion: '0.1.0', updatedAt: '2026-07-03', changelogUrl: '/changelog', recentChanges: [] }),
    }) as unknown as typeof fetch
    const v = await fetchVersion()
    expect(v.apiVersion).toBe('1.1.0')
  })
})

File: server/internal/version/version_test.go
package version

import "testing"

func TestCurrentHasRequiredFields(t *testing.T) {
	v := Current()
	if v.APIVersion == "" {
		t.Error("APIVersion empty")
	}
	if v.SkillVersion == "" {
		t.Error("SkillVersion empty")
	}
	if v.ChangelogURL == "" {
		t.Error("ChangelogURL empty")
	}
}
